.github/workflows/push-build-downstream.yaml

```diff
@@ -0,0 +1,81 @@
+name: Push Image Build Downstream
+on:
+  push:
+    branches:
+      - master
+      - release-*
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+permissions:
+  contents: read
+
+jobs:
+  push-image-to-container-registry:
+    runs-on: ubuntu-20.04
+    if: github.repository == 'red-hat-storage/rook'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+
+        # docker/setup-qemu action installs QEMU static binaries, which are used to run builders for architectures other than the host.
+      - name: set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: all
+
+      - name: log in to container registry
+        uses: docker/login-action@v3
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
```

Comment:
failing here,
```
Run docker/login-action@v3
  with:
    registry: quay.io
    ecr: auto
    logout: true
```

@Nikhil-Ladha @nigoyal@redhat.com do we need credentials separately in the rook repo?

Reply:
Yes, we need the quay credentials to push,pull the images.
I don't think there are any as of now in the repo.
@iamniting we should be saving the openshift-bot robot account's creds, right?

Reply:
We have the secrets inside the odf repo. If I am not mistaken @obnoxxx added it in the odf 3 years back. 
![Screenshot from 2024-03-20 16-59-58](https://github.com/red-hat-storage/rook/assets/22993933/54f2a26e-0700-41b6-88e6-301e2fc9c1bf)

Reply:
so we need the same secrets in the rook repo

Reply:
Yeah, seems like those are openshift robot account details only.
We can use that I think, the robot accounts are created for these purpose only.
WDYT @iamniting ?

Reply:
If you guys have permission can you add those?

Reply:
Yes, they were created for this purpose only. I do not have those details. Also, I do not have access to add them to the DS rook repo.

Reply:
I hope @agarwal-mudit have it, can you help in updating it?

Reply:
@uchapaga@redhat.com do you have those